austation/code/game/objects/items/tanks/tank_types.dm

```diff
@@ -0,0 +1,88 @@
+  max_integrity = 10
+  volume = 200000
+  no_rupture = TRUE
+
+  var/datum/gas/gas_type
+  var/gas_temp
+
+/obj/item/tank/internals/occult/Initialize()
+	..()
+
+	var/list/gas_list = list()
+
+	switch(rand(1, 100))
+		if(1 to 30) //cold
+			gas_temp = rand(0, 280)
+		if(31 to 60) //normal
+			gas_temp = 293.15
+		if(61 to 90) //hot
+			gas_temp = rand(300, 1000)
+		if(91 to 100) //S U P E R  H O T
+			gas_temp = rand(1001, 100000)
+
+	switch(rand(1, 100))
+		if(1 to 10) //better luck next time
+			gas_list = list(/datum/gas/nitrogen, /datum/gas/carbon_dioxide, /datum/gas/nitrous_oxide, /datum/gas/water_vapor)
+		if(11 to 50) //usefull but boring
+			gas_list = list(/datum/gas/oxygen)
+			gas_temp = 293.15
+		if(50 to 90) //we are getting somewhere
+			gas_list = list(/datum/gas/plasma, /datum/gas/nitryl, /datum/gas/bz, /datum/gas/miasma, /datum/gas/pluoxium)
+		if(91 to 100) //are you winning son?
+			gas_list = list(/datum/gas/tritium, /datum/gas/stimulum, /datum/gas/hypernoblium)
+
+	gas_type = pick(gas_list)
+
+/obj/item/tank/internals/occult/process()
+	..()
+
+	air_contents.clear()
+	air_contents.set_moles(gas_type, (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C))
+	air_contents.set_temperature(gas_temp)
+
+/obj/item/tank/internals/occult/deconstruct(disassembled = TRUE)
+  explosion(src, 0, 0, 8)
+  ..()
+
+
+/////////////////////  COMBAT TANK  /////////////////////
+/obj/item/tank/internals/combat  //  Contains a small measure of nitryl and oxygen, you can alt-click it to activate the nitryl secondary effects.
+	name = "combat mix tank"
+	desc = "A partial tank of nitryl and pluoxium. Alt-click to quickly shift modes."
+	icon = 'austation/icons/obj/tank.dmi'
+	icon_state = "combat"
+	distribute_pressure = 26
+	force = 10
+	dog_fashion = /datum/dog_fashion/back
+
+/obj/item/tank/internals/combat/advanced  //  Admin spawn, contains the incredibly powerful stimulum.  Do not give this to players through normal means.
+	name = "advanced combat mix tank"
+	desc = "A full tank of stimulum and pluoxium. The real deal, feel blessed."
+	distribute_pressure = 13
+	icon_state = "combat_adv"
+
+/obj/item/tank/internals/combat/populate_gas()
+	air_contents.set_moles(/datum/gas/oxygen, (0.7*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)*0.63)
+	air_contents.set_moles(/datum/gas/nitryl, (0.7*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)*0.37)  //  should give up to a maximum of 5 minutes and 15 seconds of meth speed
+
+/obj/item/tank/internals/combat/advanced/populate_gas()
+	air_contents.set_moles(/datum/gas/pluoxium, (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)*0.17)
+	air_contents.set_moles(/datum/gas/stimulum, (10*ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C)*0.83)
+
+/obj/item/tank/internals/combat/AltClick(mob/user)
+	. = ..()
+	if(istype(src, /obj/item/tank/internals/combat/advanced))  //  Advanced tanks use stimulum and don't need to switch pressures
+		return
+	if(distribute_pressure == 26)
```

Comment:
```suggestion
	if(distribute_pressure <= 26)
```

Reply:
Sorry, I don't understand your reasoning here.
26 is the minimum breathable pressure so if the pressure is below that, the tank is probably spent.
The idea is to swap from one threshold to the other and 26 is the threshold.

Reply:
now switches based on the closer threshold

Reply:
I'm now taking an entirely different approach, but if this isn't okay, I'll revert commit back to this solution

Reply:
I'm now taking an entirely different approach, but if this isn't okay, I'll revert commit back to this solution